### PR TITLE
Prompt to keep or replace Featured Image when replacing Image block media that is set as the Featured Image

### PR DESCRIPTION
**Pull request by dcalhoun**
## Related PRs

* https://github.com/WordPress/gutenberg/pull/34666
* https://github.com/wordpress-mobile/gutenberg-mobile/pull/3930
* https://github.com/wordpress-mobile/WordPress-Android/pull/15303

Relates to https://github.com/WordPress/gutenberg/issues/34429. If an Image block is set as the post's Featured Image, replacing the Image block's media will prompt the user as to whether they want to replace the Featured Image as well.

To test: See https://github.com/WordPress/gutenberg/pull/34666. 

## Regression Notes
1. Potential unintended areas of impact
    Existing Featured Image flow. 
2. What I did to test those areas of impact (or what existing automated tests I relied on)
    Verified setting and unsetting Featured Image continues to function as expected. 
3. What automated tests I added (or what prevented me from doing so)
    None. We intend to relocate this from native to JavaScript. 

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift [hunk 1]
```diff
@@ -690,24 +690,24 @@ extension GutenbergViewController: GutenbergBridgeDelegate {
             self.showAlertForReplacingFeaturedImage(mediaID: mediaID)
         }
 
-        if presentedViewController != nil {
+        if let viewController = presentedViewController, !viewController.isBeingDismissed {
```
Comment (dcalhoun) [inline WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift:707]: Now that we trigger this alert when the image is replaced, the previous code erroneously dismissed the entire editor itself. This occurred as the Media Picker modal was already actively `isBeingDismissed` when `dismiss` was invoked. By checking `isBeingDismissed`, we essentially opt to not dismiss the `presentedViewController` if it is already on its way out.

Comment (SiobhyB) [reply WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift:707]: Nice, thank you for expanding on that! Putting in my knowledge bank for the future. :)

**Review by SiobhyB — APPROVED** [review]
Approved via https://github.com/WordPress/gutenberg/pull/34666#pullrequestreview-865003162

**Comment by jkmassel** [comment]
You can test the <strong>Jetpack</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/jetpack-installable-builds/'>here</a> with build number: <code>pr17132-a681ad4</code>. IPA is available <a href='https://appcenter-filemanagement-distrib1ede6f06e.azureedge.net/ad890fe9-ff3b-4597-bc3a-6b5163dff24e/Jetpack%20Alpha.ipa?sv=2019-02-02&sr=c&sig=4Ou4I27fka7vrNXZa3%2FnBUBJ6sBD4SEOqu0zmz5PlIg%3D&se=2022-02-03T16%3A46%3A01Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

**Comment by jkmassel** [comment]
You can test the <strong>WordPress</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/WPiOS-One-Offs/'>here</a> with build number: <code>pr17132-a681ad4</code>. IPA is available <a href='https://appcenter-filemanagement-distrib3ede6f06e.azureedge.net/2c3f6771-44eb-4940-9ac7-6309024c201c/WordPress%20Alpha.ipa?sv=2019-02-02&sr=c&sig=Sc1JVRapVeJKayhzcuP6GHNbbtk9McH032wIsoEkv7I%3D&se=2022-02-03T16%3A46%3A49Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

**Pull request merged**